device.go

```diff
@@ -421,6 +443,50 @@ func getSCSIDevPath(scsiAddr string) (string, error) {
 	return filepath.Join(devPrefix, scsiDiskName), nil
 }
 
+// findBlkCCWDevPath finds the CCW block disk name associated with the given CCW block path.
+func findBlkCCWDevPath(blkCCWpath string) (string, error) {
+	files, err := ioutil.ReadDir(blkCCWpath)
+	if err != nil {
+		return "", err
+	}
+
+	for _, f := range files {
+		if strings.Contains(f.Name(), blkCCWSuffix) {
+			subPath := filepath.Join(blkCCWpath, f.Name(), "block")
+			files2, err := ioutil.ReadDir(subPath)
+			if err != nil {
+				return "", err
+			}
+			if len(files2) != 1 {
+				return "", grpcStatus.Errorf(codes.Internal,
+					"Expecting a single blk CCW device in %s found %v",
+					subPath, files2)
+			}
+			return files2[0].Name(), nil
+		}
+	}
+	return "", grpcStatus.Errorf(codes.Internal, "Path %s for blk CCW not found", blkCCWpath)
+}
+
+// getBlkCCWDevPath returns the CCW block path based on the bus ID
+func getBlkCCWDevPath(bus string) (string, error) {
```

Comment:
What if `bus` is one of the following?

- `""`
- `../../../dev` or similar?

Reply:
if it is a weird bus string like "" or "../../../dev", the function findBlkCCWDevPath won't be able to find the path and it will print the error Path %s for blk CCW not found". The expected path to find the blk ccw name is "/sys//bus/ccw/devices/<bus>/virtio<n>/block/<disk-name>". 
If the bus is "" the function is not able to find the subfolder  "/virtio<n>/block/<disk-name>" in  "/sys//bus/ccw/devices//" and it fails.

Reply:
but I can add a check for the bus format. It should always be in the format 0.<n>.<dddd>

Reply:
@jodh-intel I add the function `checkCCWBusFormat` to check the bus format and updates the tests